templates/tools/dockerfile/test/sanity/Dockerfile.template

```diff
@@ -48,9 +48,10 @@
   #======================================
   # More sanity test dependencies (bazel)
   RUN apt-get install -y openjdk-8-jdk
-  # TOOD(jtattermusch): pin the bazel version
+  # Revert Bazel to an old commit that supports running ./compile.sh 
+  # without a local protoc dependency
   RUN git clone https://github.com/bazelbuild/bazel.git /bazel
-  RUN cd /bazel && ./compile.sh
+  RUN cd /bazel && git reset --hard 19ece7c663cd99d338306e302e0f7ce023becfe3 && ./compile.sh
```

Comment:
It seems like Bazel deletes their release branches whenever they migrate to a new version, so we can't rely on checking out the release branch, but you're right about tags staying forever. I just needed to do `git checkout tags/0.4.1` to checkout the tag and not the branch.